docs/projects/2024/architecture-transition/record-summaries.md

```diff
@@ -0,0 +1,79 @@
+
+The scope of changes we're currently making to record summaries for Mathesar beta do _not_ allow users to define such queries. However the changes _do_ lay an architectural foundation for eventually moving in that direction. These future considerations do not require any additional work right now. Rather, the future considerations are informing our architectural design by nudging us toward using flexible query-based logic.
+
+
+## Old record summaries
+
+_This section reviews the behavior of record summaries as of Mathesar 0.1.7 just to provide context to this document. This behavior is specified in more detail within our [old specs document](../../../design/specs/record-summary.md)._
+
+Without any user configuration, records are summarized by their first non-PK column.
+
+Users can customize the behavior by configuring a record summary template like `{First Name} {Last Name}`. In the front end, this template is converted to string like `{817} {819}`, with the numbers representing the Django ids of column objects. That converted template is then persisted in metadata associated with a table.
+
+Any time a _foreign key column_ is referenced (either explicitly in a template or implicitly via the default template), a "transitive summary" is produced by using the record summary for the referenced table.
+
+As an example, to configure the record summary for the Books table so that books are summarized like "Blink, by Malcom Gladwell", we first need to configure the record summary template for the Authors table as `{First Name} {Last Name}` and then configure the record summary template for the Books table as `{Title}, by {Author}`
+
+## New record summaries
+
+_This section describes how we intend for record summaries to work in Mathesar's beta release._
+
+Without any user configuration, records will be summarized by their first _text-like_ column. This is a slight change. For example, if the PK is text, then we use that. If the PK is numeric and the first non-pk column is a numeric FK column, then we keep looking to see if we can find a text column.
+
+Users will still be able to customize the template, but the template DSL will have slightly new syntax. Column references within curly braces will need to be quoted in the same manner as in SQL. So `{first_name} {last_name}` will work if the columns are `first_name` and `last_name`. But if the columns are `First Name` and `Last Name`, then the template will need to be `{"First Name"} {"Last Name"}`. The same quoting and escaping rules apply as with SQL. Any references which fail to parse correctly are interpreted as string literals to be included in the template verbatim. Users should rarely be interacting with the template DSL manually because Mathesar has (and will continue to have) a button to insert a column into the template.
```

Comment:
Where do you foresee doing the translation between the "human friendly" version here and "machine friendly" storage form below? If in the front end, then I think we should consider coming up with something a bit simpler w.r.t. quoting and escaping. I'd like to avoid translating human-friendly to numeric identifiers in the DB if possible, since we can reuse some pieces from the data explorer and avoid some round-trips to the DB that way.